### ISSUE TITLE
ITV Programs Failing

### ISSUE THREAD

**Issue by AtomicLaser** [Ascoware/get-iplayer-automator#371]
*What steps will reproduce the problem? Be specific, and provide as much detail as possible.*
Download any ITV Program (I'm using Emmerdale, Coronation Street, Ant & Dec Limitless etc)

*What is the expected output? What do you see instead?*
Program downloads

*What version of GiA are you using? On what version of OS X?*
GiA v1.21.12 | macOS 12.0.1

*Are you using a web proxy? DNS proxy? VPN? VPS?*
No

*What program are you attempting to download (full title, series and episode)? **Provide only ONE program as an example.** *
Emmerdale, Coronation Street, Ant & Dec Limitless etc

*Please provide any additional information below.*
BBC programs download fine, no useable information in logs that I can see.

**Comment by montereyman** [comment]
Hi- Having exactly the same issues as AtomicLaser.
ITV programs will not download....BBC all ok to download.
No proxy or VPN on.
Running lastest version and macOS12.1

**Comment by SOLOSAIR** [comment]
agreed. my comments made in issue #363

**Comment by Mac-Man-APH** [comment]
Also same problem for me, see my comments in issue [https://github.com/Ascoware/get-iplayer-automator/issues/363](url)

**Comment by Mac-Man-APH** [comment]
So I tried again this morning and both ITV and ITV Hub episodes as those that failed previously now all downloaded successfully. I had made no changes to software or settings on my laptop. Can others please try and confirm that the problem seems to be cured.

**Comment by brgphil** [comment]
I'm finding it's hit and miss - some programs download with no issues, others fail once then download correctly, some fail a few times then eventually download.

**Comment by Lordsnooty05** [comment]
Yes, I am the same as brgphil I was trying to download the series 3 of The Bay and a couple of episodes downloaded and a couple didn’t but when I used Downie those episodes that failed all downloaded using Downie

**Comment by AtomicLaser** [comment]
I too am finding it a bit hit and miss at the moment, out of 17 in my queue (so far) only one has downloaded successfully.

**Comment by AtomicLaser** [comment]
I seem to have some success by doing the below which was referenced in another thread:

@robincm 
To quickly do this from Terminal run these commands:

```
sudo mkdir -p  ~/.config/youtube-dl
echo '--user-agent "Mozilla/5.0"' | sudo tee -a ~/.config/youtube-dl/config
```

The first command will create the youtube-dl directory if it does not already exist, and the second will append the user agent option to the config file (creating the file if it doesn't exist).

_Originally posted by @gadgetchnnel in https://github.com/Ascoware/get-iplayer-automator/issues/370#issuecomment-1019510290_

**Comment by robincm** [comment]
@AtomicLaser Thanks for that - I have picked it up on the other thread and after a couple of false starts now have it successfully downloading.

**Comment by Mac-Man-APH** [comment]
> I seem to have some success by doing the below which was referenced in another thread:
> 
> ```
> sudo mkdir -p  ~/.config/youtube-dl
> echo '--user-agent "Mozilla/5.0"' | sudo tee -a ~/.config/youtube-dl/config
> ```
> 
> The first command will create the youtube-dl directory if it does not already exist, and the second will append the user agent option to the config file (creating the file if it doesn't exist).
> 
> _Originally posted by @gadgetchnnel in [#370 (comment)](https://github.com/Ascoware/get-iplayer-automator/issues/370#issuecomment-1019510290)_

AtomicLaser thank you so much for that as it CURED the problem here! Hopefully another user will try this and let us know of their findings.

**Comment by jeanpaulz85** [comment]
Did not work for me

**Comment by WilliamOfGaunt** [comment]
Magic – it worked for me too! My thanks to the original poster. (GiA 1.21.12, MacOS 12.1)

B.

**Comment by montereyman** [comment]
Great job...ITV now all ok for downloads.

Thanks to AtomicLaser & Gadgetchnnel

**Comment by SOLOSAIR** [comment]
Compared to you guys I'm a complete novice and followed you here from the GiA main webpage a few years ago. I really don't want to use terminal, so will there be a GiA update that fixes this problem please? Still no ITV downloads here. 😟

**Comment by Lordsnooty05** [comment]
Solosair Im a bit like you in that I don’t want to mess about with Terminal but if you use a Mac look up an app called “Downie “ as it will download all the ITV programs for you until someone like Scott fixes this…..he usually gets a solution to fix it……this would just tie you over until it’s fixed

**Comment by naq59** [comment]
This trick also worked for me having had a number of issues since last week even being able to access the site, let alone download. I finally had to switch my User agent on Safari to IOS iPhone(!) to get it working. I can also confirm Downie working regardless. I can understand not wanting to do Terminal commands if you're not familiar with it but it´s maybe not quite so daunting as you might think. Just make sure you copy and paste exactly, watching out for empty spaces before commands. Thanks for the help.

**Comment by skovatch** [comment]
I can push out an update that sets the User-Agent on ITV downloads. Why that should matter is not clear to me right now, but it also can't hurt anything by setting it. 

Depending on how busy my afternoon is you should see something in a day or two.

**Comment by sleaux-meaux** [comment]
youtube-dl generates random user agent values, but the only thing that varies is the Chrome version number. It plugs in one of about 1500 possible (and very obsolete) values. Some hang and some don't. I couldn't discern a pattern. Even bogus values for `--user-agent` appear to work so long as a bad Chrome version number isn't included.

Also consider adding `--socket-timeout` just in case this problem should arise again. The default value is 600 secs, which is way too long in this context. 60 secs is probably enough.

**Comment by SOLOSAIR** [comment]
many, many thanks for the update... you have no idea how happy this has made me! 😃🎉

**Comment by skovatch** [comment]
Fixed in 1.21.13
